login/login.go

```diff
@@ -0,0 +1,390 @@
+	ExpiresIn        int    `json:"expires_in"`
+	Error            string `json:"error"`
+	ErrorDescription string `json:"error_description"`
+}
+
+type UserInfo struct {
+	Sub               string   `json:"sub"`
+	Name              string   `json:"name"`
+	PreferredUsername string   `json:"preferred_username"`
+	GivenName         string   `json:"given_name"`
+	FamilyName        string   `json:"family_name"`
+	Email             string   `json:"email"`
+	EmailVerified     bool     `json:"email_verified"`
+	Ga4ghPassportV1   []string `json:"ga4gh_passport_v1"`
+}
+
+type DeviceLogin struct {
+	BaseURL         string
+	ClientID        string
+	S3Target        string
+	PollingInterval int
+	LoginResult     *Result
+	UserInfo        *UserInfo
+	wellKnown       *OIDCWellKnown
+	deviceLogin     *DeviceLoginResponse
+}
+
+type AuthInfo struct {
+	ClientID  string `json:"client_id"`
+	OidcURI   string `json:"oidc_uri"`
+	PublicKey string `json:"public_key"`
+	InboxURI  string `json:"inbox_uri"`
+}
+
+// requests the /info endpoint to fetch the parameters needed for login
+func GetAuthInfo(baseURL string) (*AuthInfo, error) {
+	url := baseURL + "/info"
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	var result AuthInfo
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+// creates a .sda-cli-session file and updates its values
+func (login *DeviceLogin) Session() error {
+	log.Info("Creating session file")
+	file, err := os.Create(".sda-cli-session")
+	if err != nil {
+		return err
+	}
+
+	s3Config, err := login.GetS3Config()
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(file, `access_key = %v
+secret_key = %v
+access_token = %v
+host_bucket = %v
+host_base = %v
+multipart_chunk_size_mb = %v
+guess_mime_type = %v
+check_ssl_certificate = %v
+encoding = %v
+check_ssl_hostname = %v
+use_https = %v
+socket_timeout = %v
+human_readable_sizes = %v`,
+		s3Config.AccessKey, s3Config.SecretKey, s3Config.AccessToken,
+		s3Config.HostBucket, s3Config.HostBase, s3Config.MultipartChunkSizeMb,
+		s3Config.GuessMimeType, s3Config.CheckSslCertificate, s3Config.Encoding,
+		s3Config.CheckSslHostname, s3Config.UseHTTPS, s3Config.SocketTimeout,
+		s3Config.HumanReadableSizes)
+	defer file.Close()
+
+	return nil
+}
+
+// NewDeviceLogin() returns a new `DeviceLogin` with the given `url` and
+// `clientID` set.
+func NewDeviceLogin(args []string) (DeviceLogin, error) {
+
+	var url string
+	err := Args.Parse(args[1:])
+	if err != nil {
+		return DeviceLogin{}, errors.New("failed parsing arguments")
+	}
+	if len(Args.Args()) == 1 {
+		url = Args.Args()[0]
+	}
+	log.Println("url: ", url)
+	info, err := GetAuthInfo(url)
+	info.ClientID = "8b7b0168-6b16-4fd2-baec-b0a28b0d5cb0"
+	info.InboxURI = "s3.bp.nbis.se"
+	info.OidcURI = "https://login.elixir-czech.org/oidc"
+	if err != nil {
```

Comment:
```suggestion
	info, err := GetAuthInfo(url)
	if err != nil {
```
These should not be hardcoded.

Reply:
Removed, along with the debugging logs.